R/arg_reconcile.R

```diff
@@ -0,0 +1,115 @@
+#'                ignored). Arguments in \code{.args} are overridden by
+#'                arguments of the same name (if any) in \code{...}
+#' @param .args   A list or \code{alist} of named arguments, to be merged 
+#'                with \code{...}. Arguments in \code{.args} are overridden by
+#'                arguments of the same name (if any) in \code{...}
+#' @param .docall If set to \code{TRUE} will not only clean up the arguments but 
+#'                also execute \code{fn} with those arguments (\code{FALSE} by
+#'                default) and return the results
+#' @param .wtlist Whitelist. If not empty, only arguments named here will be 
+#'                permitted, and only if they satisfy the conditions implied by
+#'                the other arguments. Evaluated before \code{.remap}.
+#' @param .bklist Blacklist. If not empty, arguments named here will be removed 
+#'                even if they satisfy the conditions implied by the other 
+#'                arguments. Evaluated before \code{.remap}.
+#' @param .remap  An optional named character vector or named list of character 
+#'                values for standardizing arguments that play the same role but
+#'                have different names in different functions. Evaluated after
+#'                \code{.bklist} and \code{.wtlist}.
+#' @param .warn   Whether to issue a warning message (default) when invalid 
+#'                arguments need to be discarded.
+#' @param .error  If specified, should be the object to return in the event of
+#'                error. This object will have the error as its 
+#'                \code{error} attribute. If not specified an ordinary error is
+#'                thrown with an added hint on the documentation to read for 
+#'                troubleshooting. Ignored if \code{.docall} is \code{FALSE}.
+#'                The point of doing this is fault-tolerance-- if this function
+#'                is part of a lengthy process where you want to document an 
+#'                error but keep going, you can set \code{.error} to some object
+#'                of a compatible type. That object will be returned in the 
+#'                event of error and will have as its \code{"error"} attribute 
+#'                the error object.
+#' @param .finish A function to run on the result before returning it. Ignored 
+#'                if \code{.docall} is \code{FALSE}.
+#'
+#' @return Either a named list or the result of calling \code{fn} with the 
+#'         supplied arguments
+#'
+arg_reconcile <- function(fn, ..., .args = alist(), .docall = FALSE, 
+                          .wtlist = c(), .bklist= c(), .remap = list(), 
+                          .warn = TRUE, .error = "default", .finish = identity) {
+  # capture the formal arguments of the target function
+  frmls <- formals(fn)
+  # both freeform and an explicit list
+  args <- match.call(expand.dots = FALSE)$`...`
+  if (.docall) {
```

Comment:
Can we be consistent with the rest of the package on brackets/formatting - always use {} even on short expressions; else always on a separate line; no spaces after `(` or before `)` in condition statements. Always like:

```
if (cond) {
  # stuff
} else { 
  # other stuff
}

for (i in seq) {
  # stuff
}
```

Reply:
Done